app/services/bulk_hosts_manager.rb

```diff
@@ -0,0 +1,35 @@
+class BulkHostsManager
+  def initialize(hosts:)
+    @hosts = hosts
+  end
+
+  def build(reboot: false)
+    # returns missed hosts
+    @hosts.select do |host|
+      success = true
+      begin
+        host.built(false) if host.build? && host.token_expired?
+        host.setBuild
+        host.power.reset if host.supports_power_and_running? && reboot
+      rescue => error
+        message = _('Failed to redeploy %s.') % host
+        Foreman::Logging.exception(message, error)
+        success = false
+      end
+      !success
+    end
+  end
+
+  def rebuild_configuration
+    # returns a hash with a key/value configuration
+    all_fails = {}
+    @hosts.each do |host|
+      result = host.recreate_config
+      result.each_pair do |k, v|
+        all_fails[k] ||= []
+        all_fails[k] << host.name unless v
```

Comment:
Should this be `display_name` to work with the new display_fqdn_for_hosts setting?

Reply:
I dont see where this is defined?

```ruby
[2] pry(main)> Host.last.display_name
NoMethodError: undefined method `display_name' 

```

Reply:
https://github.com/theforeman/foreman/blob/develop/app/views/api/v2/hosts/base.json.rabl#L5